src/terminal/Process.cpp

```diff
@@ -472,8 +477,24 @@ string Process::workingDirectory() const
 		// ignore failure, and use default instead.
 		return "."s;
 	}
+#elif defined(__APPLE__)
+    try {
+        struct proc_vnodepathinfo vpi;
+
+        auto pid = tcgetpgrp(_pty.masterFd());
+
+        auto ret = proc_pidinfo(pid, PROC_PIDVNODEPATHINFO, 0, &vpi, sizeof(vpi));
```

Comment:
```suggestion
        auto const ret = proc_pidinfo(pid, PROC_PIDVNODEPATHINFO, 0, &vpi, sizeof(vpi));
```
or simply `if (functionCall(...) < 0) ...`